jolly-jellyfish/src/page_maker/views.py

```diff
@@ -1,3 +1,219 @@
-from django.shortcuts import render
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.shortcuts import get_object_or_404
+from django.urls import reverse_lazy
+from django.views.generic import (
+    TemplateView,
+    FormView,
+    DetailView,
+    UpdateView,
+    DeleteView,
+    ListView
+)
 
-# Create your views here.
+from .forms import UserRegisterForm, WebpageForm, TemplateForm, CommentForm
+from .models import User, Webpage, Template, Comment
+
+
+class MainView(TemplateView):
+    template_name = "page_maker/main.html"
+
+
+class UserRegister(FormView):
+    template_name = 'registration/register.html'
+    form_class = UserRegisterForm
+    success_url = '/users/login'
+
+    def form_valid(self, form):
+        form.save()
+        return super().form_valid(form)
+
+
+class UserDetailView(DetailView):
+    model = User
+    context_object_name = 'viewed_user'
+    template_name = 'page_maker/user_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # user = get_object_or_404(User, pk=self.kwargs.get('pk'))
+        user = self.get_object()
+        context['user_pages'] = Webpage.objects.filter(author=user)
+        return context
+
+
+class UserUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+    model = User
+    template_name = 'page_maker/user_update.html'
+    fields = ['first_name', 'last_name', 'email']
+    success_url = '/'
+
+    def test_func(self):
```

Comment:
We might want to rename this function if we can... If not, move the functionality to anther function, call it something like `request_user_is_self` (bonus points if it is a property), then call it from this?

Reply:
This is actually an overriding method thing with Django and so cannot be renamed.
![image](https://user-images.githubusercontent.com/30503695/89539988-5f5ace00-d7f4-11ea-812d-1dd15267f380.png)